pkg/migrate/helpers.go

```diff
@@ -0,0 +1,454 @@
+			if err != nil {
+				return fmt.Errorf("invalid --http-port value: %w", err)
+			}
+			parsedInput.httpPort = num
+
+		case strings.HasPrefix(arg, "--insecure"):
+			parsedInput.tlsEnabled = false
+
+		// CockroachDB Enterprise Operator automatically adds "--logs" flag if it is not present. "--logtostderr" just
+		case strings.HasPrefix(arg, "--logtostderr"):
+			continue
+
+		default:
+			if matches := flagRegex.FindStringSubmatch(arg); len(matches) == 3 {
+				flags[fmt.Sprintf("--%s", matches[1])] = matches[2]
+			}
+			parsedInput.flags = flags
+		}
+	}
+
+	// The helm chart configures crdb to listen for grpc and sql on one port and for http on another.
+	// The cloud operator uses three distinct ports for grpc, sql, and http.
+	// Default port for grpc is 26258
+	parsedInput.grpcPort = 26258
+
+	return nil
+}
+
+// parseInt32 safely converts a string to int32
+func parseInt32(value string) (int32, error) {
+	num, err := strconv.ParseInt(value, 10, 32) // Base 10, 32-bit size
+	if err != nil {
+		return 0, err
+	}
+	return int32(num), nil
+}
+
+// replaceEnvVars replaces ${VAR} or $VAR with actual values from the envVars map.
+func replaceEnvVars(input string, envVars map[string]string) string {
+	// Match both ${VAR} and $VAR patterns
+	envRegex := regexp.MustCompile(`\$\{?([\w]+)\}?`)
+
+	return envRegex.ReplaceAllStringFunc(input, func(match string) string {
+		// Extract variable name (handles both ${VAR} and $VAR)
+		varName := strings.Trim(match, "${}")
+		if val, ok := envVars[varName]; ok {
+			return val
+		}
+		return match // Return original if no replacement found
+	})
+}
+
+// ConvertSecretToConfigMap retrieves a secret and creates a ConfigMap with the same data.
+func ConvertSecretToConfigMap(ctx context.Context, clientset kubernetes.Interface, namespace, secretName string) error {
+	// Get the Secret
+	secret, err := clientset.CoreV1().Secrets(namespace).Get(ctx, secretName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get secret %s: %w", secretName, err)
+	}
+
+	// Convert Secret data to ConfigMap data
+	configMapData := make(map[string]string)
+	for key, value := range secret.Data {
+		configMapData[key] = string(value) // If encoding needed
+	}
+
+	// Create a new ConfigMap
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: namespace,
+		},
+		Data: configMapData,
+	}
+
+	// Create the ConfigMap in Kubernetes
+	_, err = clientset.CoreV1().ConfigMaps(namespace).Create(ctx, configMap, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create ConfigMap %s: %w", secretName, err)
+	}
+
+	fmt.Println("ConfigMap created successfully:", secretName)
+	return nil
+}
+
+// updatePublicService updates the "cockroachdb-public" service with separate sql and grpc ports.
+func updatePublicService(ctx context.Context, clientset kubernetes.Interface, namespace, name string) error {
```

Comment:
* Remind me what the public svc is used for? Could updating it break client applications using the cluster?
* Since we're modifying an existing k8 resource, wdyt about backing up the original to disk before the update?

Reply:
Helm chart crdb statefulset runs sql and grpc both on 26257 port. The cockroachdb enterprise operator can not run both on 26257 port. Hence I am changing the grpc port to 26258 during migration. It will not break the sql connections but grpc commands needs to be run on 26258.

The public service is used to connect to cockroach sql or cockroach UI by users.

Yes. we can back it up to disk.

Reply:
Instead of updating the service, I have generated the svc manifest. Added instruction in README.md to update the service.